### PR TITLE
Bookstore: use Redux in React components.

### PR DESCRIPTION
**Pull request by bill7pearl**
# In this PR I have:
- Connected the store with the main component.
- Defined a state slice which will contain the books.
- Displayed books, received from the slice, in a reusable component.
- Dispatched actions using useDispatch.
- Added a button component which includes an event handler that adds a book to the books array (with attributes id, title and author) & another delete button with an event handler to remove the book from the array by id.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
+  {
+    itemId: id(), title: 'The World as Will and Representation', author: 'Arthur Schopenhauer', category: 'philosophy',
   },
-});
+  {
+    itemId: id(), title: 'Thus Spoke Zarathustra', author: 'Friedrich Nietzsche', category: 'philosophy',
+  },
+  {
+    itemId: id(), title: 'Les Misérables', author: 'Victor Hugo', category: 'historical fiction',
+  },
```
Comment (Qoosim) [inline src/redux/books/books.js:30]: - [x] Kindly create a variable called `initialState` and assign your array of objects to it. You can then reference to it inside your booksReducer like this :point_down: 

```
    const booksReducer = (state = initialState, action) => {
         switch (action.type) {
          ........
          }
     }
``` 
The reason for this is to make your code clean and to follow best practices.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
+], action = {}) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.book];
+    case REMOVE_BOOK:
+      return [...state.filter((book) => book.itemId !== action.itemId)];
+    default:
+      return state;
+  }
+};
```
Comment (Qoosim) [inline src/redux/books/books.js]: - [x] You're also expected to create a component for category and write the similar functions and reducer for it such that when the `check status` button is clicked, `Under Construction` text would be displayed.
You can check project requirements to confirm this. 

project-rqt

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @bill7pearl, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 -  No linter errors. :heavy_check_mark: 
 -  Good README file. :heavy_check_mark: 
 -  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (bill7pearl) [reply src/redux/books/books.js:30]: done

File: src/redux/books/books.js (hunk 2 above)
Comment (bill7pearl) [reply src/redux/books/books.js]: done

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @bill7pearl 👋 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**